src/astro/databases/base.py

```diff
@@ -227,7 +258,7 @@ def export_table_to_pandas_dataframe(self, source_table: Table) -> pd.DataFrame:
             return pd.read_sql(
                 # We are avoiding SQL injection by confirming the table exists before this statement
                 f"SELECT * FROM {table_qualified_name}",  # skipcq BAN-B608
-                con=self.sqlalchemy_engine,
+                con=self.get_sqlalchemy_engine(table=source_table),
```

Comment:
I suggest rolling back this since this function did not change the original behavior.

Reply:
It does change the behavior depending on the database. If it's postgres for example we use the database and schema of the table instead of the database class